### PR TITLE
Error message improvement

### PR DESCRIPTION
**Pull request by rpryzant**
See issue https://github.com/tensorflow/tensorflow/issues/7675

### CODE REVIEW

**Comment by rpryzant** [comment]
I signed it!

File: tensorflow/python/ops/rnn.py [hunk 1]
```diff
@@ -515,7 +515,7 @@ def dynamic_rnn(cell, inputs, sequence_length=None, initial_state=None,
       state = initial_state
     else:
       if not dtype:
-        raise ValueError("If no initial_state is provided, dtype must be.")
+        raise ValueError("A dtype must be specified if no initial_state is provided.")
```
Comment (caisq) [inline tensorflow/python/ops/rnn.py]: style nit: This line is now > 80 chars long. Can you please break the line at the first double quote and use a hanging indent of four space related to the previous line.

**Review by caisq — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/rnn.py (hunk 1 above)
Comment (caisq) [inline tensorflow/python/ops/rnn.py]: But the rewording looks good. Thanks!

Comment (rpryzant) [inline tensorflow/python/ops/rnn.py]: Done!

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Review by caisq — APPROVED** [review]

**Pull request merged**